gen/gen.go

```diff
@@ -128,18 +128,47 @@ func Gen(exercise string, j interface{}, t *template.Template) error {
 		Version: commonMetadata.Version,
 	}, j}
 
+	// Use two potential output files, one for success, and one for format.Source failure.
+	casesFileName := filepath.Join(dirExercise, "cases_test.go")
+	casesErrorFileName := filepath.Join(dirExercise, "cases_test.go-error")
+	// Begin by removing the two potential output files.
+	// When an error occurs in template Execute, there will be a missing cases_test.go.
+	// When an error occurs in source Format, there will be a missing cases_test.go,
+	// but a cases_test.go-error ( in the .gitignore BTW ) is left with the error inside
+	// allowing an examination of the generated source which has a syntax problem.
+	// When all goes successfully, cases_test.go is regenerated anew and any cases_test.go-error
+	// is removed, cleaning up a prior run which had an error.
+	//
+	_ = os.Remove(casesFileName)
+	_ = os.Remove(casesErrorFileName)
+
 	// render the Go test cases
 	var b bytes.Buffer
 	if err := t.Execute(&b, &d); err != nil {
+		log.Print("Error during template Execute")
 		return err
 	}
 	// clean it up
-	src, err := format.Source(b.Bytes())
+	srcBuf := b.Bytes()
+	src, err := format.Source(srcBuf)
 	if err != nil {
+		log.Print("Error during format.Source")
+		b.Reset()
+		_, _ = b.Write(srcBuf)
+		_, _ = b.Write([]byte("// !NOTE: Error during source formatting: Line:Column " + fmt.Sprint(err) + "\n"))
+		src = b.Bytes()
+		// Remove any stale cases_test.go
+		log.Print("For details, examine file ", casesErrorFileName)
+		// Save the raw unformatted, error-containing source for purposes of debugging the generator.
+		_ = ioutil.WriteFile(casesErrorFileName, src, 0666)
 		return err
 	}
 	// write output file for the Go test cases.
-	return ioutil.WriteFile(filepath.Join(dirExercise, "cases_test.go"), src, 0666)
+	ioerr := ioutil.WriteFile(casesFileName, src, 0666)
+	if err != nil {
```

Comment:
Ah, you are correct.  I think this logic applied with an earlier version when the coded did not return in the section prior, but now it doesn't help at all.  I will revert that change.